pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -130,8 +130,9 @@ func (m *Multicluster) AddMemberCluster(clients kubelib.Client, clusterID string
 	m.remoteKubeControllers[clusterID] = &remoteKubeController
 	m.m.Unlock()
 
+	// Only need to add service handler for kubernetes registry as `initRegistryEventHandlers`,
+	// because when endpoints update `XDSUpdater.EDSUpdate` has already been called.
```

Comment:
I guess this was added for workloads selecting pods to reconcile foreign registry instances. @rshriram ?

Reply:
Oh yes.. please dont remove this.. This is for workload entry

Reply:
nope, foreign service instance is not supported yet in multi-cluster model.

Reply:
I see. This is only for multi cluster. Missed that part.

Reply:
Ideally all endpoint changes would use EDSUpdate - even for WorkloadEntry/ServiceEntry

Reply:
Yeah, and after #25502 it is possible